tools/check-upcoming-maintenance.py

```diff
@@ -0,0 +1,130 @@
+    def get_upcoming_maint(self) -> List[str]:
+        cmd = "gcloud alpha compute instances list --project={} --filter='upcomingMaintenance:*".format(self.project)
+        if self.zones is not None:
+            # NOTE: This link will likely need to be updated once this feature moves out of alpha
+            # Used instead of --zones as that breaks the filter feature
+            links = ["\"https://www.googleapis.com/compute/alpha/projects/{}/zones/{}\"".format(self.project, x) for x in self.zones]
+            cmd += " AND (zone=" + " OR ".join(links) + ")"
+        cmd +=  "' --format='table(name, upcomingMaintenance.startTimeWindow.earliest:label=EARLIEST_START," \
+                " upcomingMaintenance.startTimeWindow.latest:label=LATEST_START," \
+                " upcomingMaintenance.canReschedule, upcomingMaintenance.type)'"
+
+        if self.screen_reader:
+            cmd = self.remove_accessibility(cmd)
+
+        res = subprocess.run(cmd, shell=True, capture_output=True, text=True)
+        if res.returncode == 0:
+            self.upc_maint = [x.split() for x in res.stdout.split("\n")[1:-1]]
+            if self.regex:
+                self.upc_maint = list(filter(lambda x: self.regex.match(x[0]), table))
+        else:
+            raise Exception("Error getting upcoming maintenance list:\n" + res.stderr)
+
+    def start_maint(self, project: str, zones: str):
+        pass
+
+    def print_maintenance_nodes(self):
+        if not self.maint_nodes or len(self.maint_nodes) == 0:
+            print("No nodes with periodic maintenance\n")
+            return
+
+        print("Nodes with PERIODIC maintenance")
+        for x in self.maint_nodes:
+            print(x)
+        print()
+
+    def print_upcoming_maintenance(self):
+        if not self.upc_maint or len(self.upc_maint) == 0:
+            print("No upcoming maintenance\n")
+            return 
+
+        print("Upcoming maintenance:")
+        row_format ="{:30}" * (len(self.upc_maint[0]))
+        print(row_format.format(*["Name", "Earliest Start", "Latest Start", "Can Reschedule", "Maintenance Type"]))
+        for row in self.upc_maint:
+            print(row_format.format(*row))
+        print()
+
+if __name__ == "__main__":
+    parser = OptionParser()
+    parser.add_option("-p", "--project", dest="project",
+                      help="GCP project id")
+    parser.add_option("-z", "--zones", dest="zones",
+                      help="deployment zones (comma delimited, no spaces)")
+    parser.add_option("-n", "--node-regex", dest="node_regex",
+                      help="Regular expression search for nodes")
+    # parser.add_option("-m", "--check-maintenance", action="store_true", dest="check_maintenance",
+    #                   help="Get nodes that have periodic maintenance setup (default)")
+    # parser.add_option("-u", "--check-upcoming", action="store_true", dest="check_upcoming",
+    #                   help="Get nodes that have upcoming maintenance")
```

Comment:
What is the plan with these commented out lines?

Reply:
Uncommented and in use.